test/code_transformation/differentiable.jl

```diff
@@ -112,4 +112,10 @@ skip_line_info(ex) = ex
     @test unionise(:(@eval foo)) ≃ unionise_macro_eval(:(@eval foo))
     @test unionise(:(@eval DiffBase foo)) ≃ unionise_macro_eval(:(@eval DiffBase foo))
     @test unionise(:(struct Foo{T<:V} end)) == unionise_struct(:(struct Foo{T<:V} end))
+
+    # @unionise with default values and keywords
+    UT = unionise_type(:T)
+    raw = :(@unionise f(x::T, y::T=2; z::T=4) = x + y + z)
+    new = :(f(x::$UT, y::$UT=2; z::T=4) = x + y + z)
+    @test macroexpand(@__MODULE__, raw) ≃ new
```

Comment:
Is there a reason to use `macroexpand` and not do this?
```julia
raw = unionise(:(f(x::T, y::T=2; z::T=4) = x + y + z))
new = :(f(x::$UT, y::$UT=2; z::T=4) = x + y + z)
@test raw ≃ new
```

Reply:
The reason is that I forgot there was a `unionise` function.